plugins/modules/database/influxdb/influxdb_retention_policy.py

```diff
@@ -272,30 +287,58 @@ def alter_retention_policy(module, client, retention_policy):
     module.exit_json(changed=changed)
 
 
+def drop_retention_policy(module, client):
+    database_name = module.params['database_name']
+    policy_name = module.params['policy_name']
+
+    changed = False
+
+    if not module.check_mode:
+        try:
+            client.drop_retention_policy(policy_name, database_name)
+        except exceptions.InfluxDBClientError as e:
+            module.fail_json(msg=e.content)
+        changed = True
+    module.exit_json(changed=changed)
```

Comment:
In check mode, you must also return `changed=True`:
```suggestion
    module.exit_json(changed=True)
```